lib/base_helper.rb

```diff
@@ -137,21 +137,32 @@ def capture_error
       #   strictly enforced, use other timeout techniques to avoid freeze
       # @param seconds [Numeric] the max number of seconds to try operation to
       #   succeed
-      # @param interval [Numeric] the interval to wait between attempts
+      # @param interval [Numeric] the interval to wait between a
+      # @param stats [Hash] collect stats
+      # @param jitter_muliplier [Numeric] use decorrelated jitter to reduce contention
+      #   between clients, see https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
       # @yield block the block will be yielded until it returns true or timeout
       #   is reached
-      def wait_for(seconds, interval: 1, stats: nil)
+      def wait_for(seconds, interval: 1, stats: nil, jitter_muliplier: 3)
         if seconds > 60
           Kernel.puts("waiting for operation up to #{seconds} seconds..")
         end
         iterations = 0
 
         start = monotonic_seconds
+        # precompute deadline
+        deadline = start + seconds
         success = false
-        until monotonic_seconds - start > seconds
+        base_interval = interval
+        until monotonic_seconds > deadline
           iterations += 1
           success = yield and break
+          if jitter_muliplier > 0
+            # cap max wait at seconds remaining
+            interval = [(deadline - monotonic_seconds), rand(base_interval...(jitter_muliplier * interval))].min
+          end
           sleep interval
+          Kernel.puts("slept #{interval} seconds..")
```

Comment:
can't seem to use `logger.debug` here, so I haven't been able to capture the sleep values.